binary_search_tree/binary_search_tree.py

```diff
@@ -5,10 +5,36 @@ def __init__(self, value):
     self.right = None
 
   def insert(self, value):
-    pass
+    if value < self.value:
+      if not self.left:
+        self.left = BinarySearchTree(value)
+      else:
+        self.left.insert(value)
+    else:
+      if not self.right:
+        self.right = BinarySearchTree(value)
+      else:
+        self.right.insert(value)
 
   def contains(self, target):
-    pass
+    if self.value == target:
+      return True
+    elif target < self.value:
+      if not self.left:
+        return False
+      else:
+        return self.left.contains(target)
+    elif target > self.value:
+      if not self.right:
+        return False
+      else:
+        return self.right.contains(target)
+    else:
+      return False
```

Comment:
This setup gets the job done pretty well. Is there a way you could reduce the runtime complexity while maintaining functionality?